source/documentation/deploying_services/s3.md

```diff
@@ -0,0 +1,154 @@
+
+#### Provision a public AWS S3 bucket
+
+You can provision an S3 buckets so that bucket is publicly accessible to everyone through the internet.
+
+Possible use cases for public S3 buckets include:
+
+- storing static objects that you do not want to include with your app
+- hosting `404` error pages that work independently of your app
+
+Run the following in the command line to provision a public AWS S3 bucket:
+
+```
+cf create-service aws-s3-bucket default SERVICE_NAME -c '{"public_bucket":true}'
+```
+where `SERVICE_NAME` is a unique descriptive name for this S3 bucket.
+
+### Bind an AWS S3 bucket to your app
+
+You must bind the S3 bucket to your app so you can get credentials to use the [AWS S3 API](https://docs.aws.amazon.com/cli/latest/reference/s3api/) to read from and write to the bucket.
+
+You can bind an S3 bucket using the command line or the manifest. We recommend that you use the command line because you can specify the permissions for the bucket.
+
+Run the following to bind the S3 bucket to your app, selecting the appropriate permission level:
+
+```
+cf bind-service APP_NAME SERVICE_NAME -c '{"permissions": "PERMISSION"}'
+```
+
+where:
+
+- `APP_NAME` is the name of a deployed instance of your app, as specified in your app's manifest or push command
+- `SERVICE_NAME` is a unique descriptive name for this S3 bucket
+- `PERMISSION` is the permission level and can be `read-write` or `read-only`
+
+You can use the [app's manifest](https://docs.cloud.service.gov.uk/deploying_apps.html#deploying-public-apps) to bind an S3 bucket to the app with default read and write permissions only. It will bind automatically when you next deploy your app. An example manifest:
+
+```
+--
+applications:
+- name: APP_NAME
+  services:
+  - SERVICE_NAME
+```
+
+### Connect to an S3 bucket from your app
+
+Restart your app to populate your `VCAP_SERVICES` [environment variable](/deploying_apps.html#system-provided-environment-variables) with the credentials needed to connect to the S3 bucket.
+
+An example `VCAP_SERVICES` environment variable:
+
+```
+{
+ "VCAP_SERVICES": {
+  "aws-s3-bucket": [
+   {
+    "binding_name": null,
+    "credentials": {
+     "aws_access_key_id": "XXXXXXXXXXXXXXXXXXXX",
+     "aws_region": "eu-west-2",
+     "aws_secret_access_key": "XXXX/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+     "bucket_name": "paas-s3-broker-prod-lon-XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
+     "deploy_env": ""
+    },
+    "instance_name": "my-bucket",
+    "label": "aws-s3-bucket",
+    "name": "my-bucket",
+    "plan": "default",
+    "provider": null,
+    "syslog_drain_url": null,
+    "tags": [
+     "s3"
+    ],
+    "volume_mounts": []
+   }
+  ]
+ }
+}
+```
+You must pass credentials from your `VCAP_SERVICES` environment variable to your AWS software development kit (SDK). Use the appropriate [Amazon SDK documentation](https://aws.amazon.com/developer/tools/#sdks) to do this.  
+
+<h2 id="amend-the-service">Use the service</h2>
```

Comment:
Should this maybe be `id="use-the-service"`?

Reply:
agree